Lesson6.ipynb

```diff
@@ -27,7 +34,686 @@
+    "#find BMI\n",
+    "\n",
+    "np_heights = np.array(heights)\n",
+    "np_weights = np.array(weights)\n",
+    "\n",
+    "bmi = np_weights/ np_heights ** 2\n",
+    "\n",
+    "print(bmi)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Let's practice with numpy array operations:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 12,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "no_of_seats_per_class = [100, 80, 60, 90] #list of seats available per class\n",
+    "no_of_students_per_class = [82, 34, 49, 88] #list of students attended per class\n",
+    "\n",
+    "#find no. of seats vaccant per class using numpy array operations\n",
+    "#START HERE"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Yes, Numpy makes list operations less expensive and more efficient.** \n",
+    "\n",
+    "Keep in mind, the same applies for arthematic operators too. For example, "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 17,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "py_list = [1, 2, 3]\n",
+    "print(\"sum of lists:\",py_list + py_list)               # results in sum of elements as the type is list. where as,\n",
+    "\n",
+    "np_list = np.array(py_list)\n",
+    "print(\"sum of np arrays:\", np_list+np_list)                 #results in element wise sum of the arrays"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "#### Numpy Subsetting \n",
+    "\n",
+    "\n",
+    "**slicing** \n",
+    "\n",
+    "\n",
+    "Basic slicing is an extension of Python's basic concept of slicing to n dimensions. A Python slice object is constructed by giving a **start**, **stop**, and **step** parameters to the **built-in slice function**. This slice object is passed to the array to extract a part of an array.\n",
+    "\n",
+    "*`sliceObject = slice(start, stop, step)`*\n",
+    "\n",
+    "*`arrayObject[sliceObject]`*\n",
+    "\n",
+    "alternatively, we can use the slicing parameters seperated by colons.\n",
+    "\n",
+    "*`arrayObject[start:stop:step]`*\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 28,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[ 1  2  3  4  5  6  7  8  9 10 11 12 13 14 15 16 17 18 19]\n",
+      "Using slice function:  [3 7]\n",
+      "Using slice parameters:  [3 7]\n"
+     ]
+    }
+   ],
+   "source": [
+    "arrayObject = np.arange(1,20) # arange method will generate values within the interval. syntax arange(start,stop,step)\n",
+    "print(a)\n",
+    "sliceObject = slice(2,10,4)\n",
+    "\n",
+    "sliceArray = arrayObject[sliceObject]\n",
+    "print(\"Using slice function: \", sliceArray)\n",
+    "\n",
+    "paramArray = arrayObject[2:10:4]\n",
+    "print(\"Using slice parameters: \", paramArray)"
+   ]
```

Comment:
This code cell fails: `NameError: name 'a' is not defined`
Do you want to remove the `print(a)` method or replace `a` by `arrayObject` ?